bindings-python/gci/componentmodel.py

```diff
@@ -316,6 +329,8 @@ def from_dict(component_descriptor_dict: dict):
                 ]
             )
         )
+        if validate:
```

Comment:
let us pass-in the valide-arg to the validate-method
+ how about switching to an enum `ValidationMode` (values: IGNORE, WARN, FAIL)